handler/posts.go

```diff
@@ -0,0 +1,237 @@
+	}
+
+	p, err := s.db.Posts.Create(storage.Post{
+		Body:    form.Body,
+		Title:   form.Title,
+		OwnerID: claims.UserID,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return c.Status(http.StatusCreated).JSON(p)
+}
+
+func (s PostStorage) Update(c *fiber.Ctx) error {
+	form := UpdatePatchForm{}
+
+	if err := c.BodyParser(&form); err != nil {
+		return c.Status(http.StatusBadRequest).JSON(app.Err(err.Error()))
+	}
+	if err := Validate(&form); err != nil {
+		return c.Status(http.StatusBadRequest).JSON(app.Err(err.Error()))
+	}
+
+	token := strings.Split(c.Get("Authorization"), "Bearer ")[1]
+
+	tokenx, err := jwt.ParseWithClaims(token, &j.Claims{}, func(token *jwt.Token) (interface{}, error) {
+		return s.Secret, nil
+	})
+
+	if err != nil {
+		return c.Status(http.StatusUnauthorized).JSON(app.Err(err.Error()))
+	}
+
+	if !tokenx.Valid {
+		return errors.New("handler: token is not valid")
+	}
+
+	claims, ok := tokenx.Claims.(*j.Claims)
+
+	if !ok {
+		return errors.New("handler: claims ne ok")
+	}
+
+	post, err := s.db.Posts.Update(storage.Post{
+		OwnerID: claims.UserID,
+		Title:   form.Title,
+		Body:    form.Body,
+		ID:      form.ID,
+	})
+
+	if err != nil {
+		return c.Status(http.StatusNotAcceptable).JSON(app.Err(err.Error()))
+	}
+
+	return c.Status(http.StatusOK).JSON(post)
+
+}
+
+func (s PostStorage) Delete(c *fiber.Ctx) error {
+	id, err := strconv.Atoi(c.Params("id"))
+
+	if err != nil {
+		return c.Status(http.StatusBadRequest).JSON(app.Err(err.Error()))
+	}
+
+	token := strings.Split(c.Get("Authorization"), "Bearer ")[1]
+
+	tokenx, err := jwt.ParseWithClaims(token, &j.Claims{}, func(token *jwt.Token) (interface{}, error) {
+		return s.Secret, nil
+	})
+
+	if err != nil {
+		return c.Status(http.StatusUnauthorized).JSON(app.Err(err.Error()))
+	}
+
+	if !tokenx.Valid {
+		return errors.New("handler: token is not valid")
+	}
+
+	claims, ok := tokenx.Claims.(*j.Claims)
+
+	if !ok {
+		return c.Status(http.StatusBadRequest).JSON(app.Err("claims is not ok"))
+	}
+
+	err = s.db.Posts.Delete(storage.Post{
+		OwnerID: claims.UserID,
+		ID:      id,
+	})
+
+	if err != nil {
+		return c.Status(http.StatusBadRequest).JSON(app.Err(err.Error()))
+	}
+
+	return c.Status(http.StatusOK).JSON(app.Ok())
+
+}
+
+func (s PostStorage) ByID(c *fiber.Ctx) error {
+	id, err := strconv.Atoi(c.Params("id"))
+
+	if err != nil {
+		return c.Status(http.StatusBadRequest).JSON(app.Err(err.Error()))
+	}
+
+	post, err := s.db.Posts.ByID(id)
+
+	if err != nil {
+		return c.Status(http.StatusNotFound).JSON(app.Err(err.Error()))
+	}
+
+	return c.Status(http.StatusOK).JSON(post)
+}
+
+func (s PostStorage) PostsByIDs(c *fiber.Ctx) error {
+	ids := strings.Split(c.Params("ids"), ",")
+
+	var posts []storage.Post
+
+	for _, id := range ids {
+		id, err := strconv.Atoi(id)
+
+		if err != nil {
+			return c.Status(http.StatusBadRequest).JSON(app.Err(err.Error()))
+		}
+
+		post, err := s.db.Posts.ByID(id)
+
+		if err != nil {
+			continue
```

Comment:
You should check here if err is sql.ErrNoRows, if it is, then continue else return

Reply:
You should'nt make for loop for this, this is inefficient. Make 1 request isntead for loop. Example:
```
func (db *DB) PostsIn(ids []int) (posts []Post, err error) {
	const q = `
		SELECT
			*
		FROM posts
		WHERE id IN (?)`

	query, args, err := sqlx.In(q, ids)
	if err != nil {
		return nil, err
	}

	return users, db.Select(&posts, db.Rebind(query), args...)
}
```